libs/agno/agno/agent/remote.py

```diff
@@ -0,0 +1,358 @@
+        return config
+
+    @cached_property
+    def name(self) -> Optional[str]:
+        if self._agent_config is not None:
+            return self._agent_config.name
+        return self.agent_id
+
+    @cached_property
+    def description(self) -> Optional[str]:
+        if self._agent_config is not None:
+            return self._agent_config.description
+        return ""
+
+    @cached_property
+    def role(self) -> Optional[str]:
+        if self._agent_config is not None:
+            return self._agent_config.role
+        return None
+
+    @cached_property
+    def tools(self) -> Optional[List[Dict[str, Any]]]:
+        if self._agent_config is not None:
+            return json.loads(self._agent_config.tools["tools"]) if self._agent_config.tools else None
+        return None
+
+    @cached_property
+    def db(self) -> Optional[RemoteDb]:
+        if self._agent_config is not None and self._agent_config.db_id is not None:
+            config = self._config
+            db_id = self._agent_config.db_id
+            session_table_name = None
+            knowledge_table_name = None
+            memory_table_name = None
+            metrics_table_name = None
+            eval_table_name = None
+            traces_table_name = None
+            if config and config.session and config.session.dbs is not None:
+                session_dbs = [db for db in config.session.dbs if db.db_id == db_id]
+                session_table_name = session_dbs[0].tables[0] if session_dbs and session_dbs[0].tables else None
+            if config and config.knowledge and config.knowledge.dbs is not None:
+                knowledge_dbs = [db for db in config.knowledge.dbs if db.db_id == db_id]
+                knowledge_table_name = knowledge_dbs[0].tables[0] if knowledge_dbs and knowledge_dbs[0].tables else None
+            if config and config.memory and config.memory.dbs is not None:
+                memory_dbs = [db for db in config.memory.dbs if db.db_id == db_id]
+                memory_table_name = memory_dbs[0].tables[0] if memory_dbs and memory_dbs[0].tables else None
+            if config and config.metrics and config.metrics.dbs is not None:
+                metrics_dbs = [db for db in config.metrics.dbs if db.db_id == db_id]
+                metrics_table_name = metrics_dbs[0].tables[0] if metrics_dbs and metrics_dbs[0].tables else None
+            if config and config.evals and config.evals.dbs is not None:
+                eval_dbs = [db for db in config.evals.dbs if db.db_id == db_id]
+                eval_table_name = eval_dbs[0].tables[0] if eval_dbs and eval_dbs[0].tables else None
+            if config and config.traces and config.traces.dbs is not None:
+                traces_dbs = [db for db in config.traces.dbs if db.db_id == db_id]
+                traces_table_name = traces_dbs[0].tables[0] if traces_dbs and traces_dbs[0].tables else None
+            return RemoteDb(
```

Comment:
Would make sense for RemoteDb to know how to init itself given the config, to encapsulate this logic there

```python
return RemoteDb(self._agent_config)
```